packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -491,18 +468,118 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
-        child: Container(
-          key: _fieldKey,
-          child: Shortcuts(
-            shortcuts: _shortcuts,
-            child: Actions(
-              actions: _actionMap,
-              child: CompositedTransformTarget(
-                link: _optionsLayerLink,
-                child: fieldView,
+        child: LayoutBuilder(
+          builder: (BuildContext context, BoxConstraints boxConstraints) {
+            _fieldBoxConstraints.value = boxConstraints;
+            return Shortcuts(
+              shortcuts: _shortcuts,
+              child: Actions(
+                actions: _actionMap,
+                child: CompositedTransformTarget(
+                  link: _optionsLayerLink,
+                  child: fieldView,
+                ),
               ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+class _OptionsView<T extends Object> extends StatefulWidget {
+  const _OptionsView({
+    required this.fieldBoxConstraints,
+    required this.highlightedOptionIndex,
+    required this.layerLink,
+    required this.openDirection,
+    required this.options,
+    required this.optionsViewBuilder,
+    required this.select,
+  });
+
+  final ValueNotifier<BoxConstraints> fieldBoxConstraints;
+  final ValueNotifier<int> highlightedOptionIndex;
+  final LayerLink layerLink;
+  final OptionsViewOpenDirection openDirection;
+  final AutocompleteOptionsViewBuilder<T> optionsViewBuilder;
+  final AutocompleteOnSelected<T> select;
+  final Iterable<T> options;
+
+  @override
+  State<_OptionsView<T>> createState() => _OptionsViewState<T>();
+}
+
+class _OptionsViewState<T extends Object> extends State<_OptionsView<T>> {
+  void onFieldBoxConstraintsChange() {
+    SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
+      if (!mounted) {
+        return;
+      }
+      setState(() {});
+    }, debugLabel: 'RawAutocomplete.onFieldBoxConstraintsChange');
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    widget.fieldBoxConstraints.addListener(onFieldBoxConstraintsChange);
+  }
+
+  @override
+  void dispose() {
+    widget.fieldBoxConstraints.removeListener(onFieldBoxConstraintsChange);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final TextDirection textDirection = Directionality.of(context);
+    final Alignment followerAlignment = switch (widget.openDirection) {
+      OptionsViewOpenDirection.up => AlignmentDirectional.bottomStart,
+      OptionsViewOpenDirection.down => AlignmentDirectional.topStart,
+    }.resolve(textDirection);
+    final Alignment targetAnchor = switch (widget.openDirection) {
+      OptionsViewOpenDirection.up => AlignmentDirectional.topStart,
+      OptionsViewOpenDirection.down => AlignmentDirectional.bottomStart,
+    }.resolve(textDirection);
+
+    return Positioned(
+      width: widget.fieldBoxConstraints.value.maxWidth,
+      // TODO(justinmc): If I don't give a height, height constraints are
```

Comment:
IIRC the Overlay does not give its children unbounded constraints (it should at least be bounded by the physical size of the screen). Do you mean a tight height constraint or a loose height constraint?

Reply:
Loose: 0.0<=h<=Infinity

Reply:
Without the Positioned widget the constraints are tight to the size of the screen.

Reply:
Ahhh I see, for positioned children it's not bounded. But why didn't the options view child no like `0.0<=h<=Infinity`? Because it's using `Align` so it doesn't know how to vertically align with infinite amount of vertical space?

Reply:
Thanks, that seems to work!

[The test that failed](https://github.com/flutter/flutter/blob/52961f761f80f602375dd8b7f3f08335d4166c66/packages/flutter/test/widgets/autocomplete_test.dart#L132) has a ListView in the optionsViewBuilder, which I think was the problem.